src/Nethermind/Nethermind.Consensus/Validators/TxValidator.cs

```diff
@@ -2,294 +2,450 @@
-                {
-                    if (wrapper.Blobs[i].Length != Ckzg.Ckzg.BytesPerBlob)
-                    {
-                        error = TxErrorMessages.ExceededBlobSize;
-                        return false;
-                    }
-                    if (wrapper.Commitments[i].Length != Ckzg.Ckzg.BytesPerCommitment)
-                    {
-                        error = TxErrorMessages.ExceededBlobCommitmentSize;
-                        return false;
-                    }
-                    if (wrapper.Proofs[i].Length != Ckzg.Ckzg.BytesPerProof)
-                    {
-                        error = TxErrorMessages.InvalidBlobProofSize;
-                        return false;
-                    }
-                }
-
-                Span<byte> hash = stackalloc byte[32];
-                for (int i = 0; i < blobCount; i++)
-                {
-                    if (!KzgPolynomialCommitments.TryComputeCommitmentHashV1(
-                            wrapper.Commitments[i].AsSpan(), hash) ||
-                        !hash.SequenceEqual(transaction.BlobVersionedHashes[i]))
-                    {
-                        error = TxErrorMessages.InvalidBlobCommitmentHash;
-                        return false;
-                    }
-                }
-
-                if (!KzgPolynomialCommitments.AreProofsValid(wrapper.Blobs, wrapper.Commitments, wrapper.Proofs))
-                {
-                    error = TxErrorMessages.InvalidBlobProof;
-                    return false;
-                }
+        return true;
+    }
+}
 
-            }
+public abstract class BaseSignatureTxValidator : ITxValidator
+{
+    protected virtual bool AdditionalCheck(Transaction transaction, IReleaseSpec releaseSpec) => false;
+
+    public bool IsWellFormed(Transaction transaction, IReleaseSpec releaseSpec, [NotNullWhen(false)] out string? error)
+    {
+        error = null;
+
+        Signature? signature = transaction.Signature;
+        if (signature is null)
+        {
+            error = TxErrorMessages.InvalidTxSignature;
+            return false;
+        }
+
+        UInt256 sValue = new(signature.SAsSpan, isBigEndian: true);
+        UInt256 rValue = new(signature.RAsSpan, isBigEndian: true);
+
+        if (sValue.IsZero || sValue >= (releaseSpec.IsEip2Enabled ? Secp256K1Curve.HalfNPlusOne : Secp256K1Curve.N))
+        {
+            error = TxErrorMessages.InvalidTxSignature;
+            return false;
+        }
 
+        if (rValue.IsZero || rValue >= Secp256K1Curve.NMinusOne)
+        {
+            error = TxErrorMessages.InvalidTxSignature;
+            return false;
+        }
+
+        if (signature.V is 27 or 28)
+        {
+            return true;
+        }
+
+        if (AdditionalCheck(transaction, releaseSpec))
+        {
             return true;
         }
+
+        if (releaseSpec.ValidateChainId)
+        {
+            error = TxErrorMessages.InvalidTxSignature;
+            return false;
+        }
```

Comment:
Instead of `AdditionalCheck` have a `virtual ValidateChainId` method, that you will override in `LegacySignatureTxValidator`

Reply:
Renamed `AdditionalCheck` to `ValidateChainId`